igvc_ws/src/igvc_serial/src/serial_node.py

```diff
@@ -0,0 +1,66 @@
+#!/usr/bin/env python
+
+import rospy
+import json
+import threading
+import serial
+from std_msgs.msg import String
+from igvc_msgs.msg import motors
+
+serials = {}
+
+class SerialReadThread(threading.Thread):
+    def __init__(self, serial_obj, topic):
+        threading.Thread.__init__(self)
+
+        self.serial_obj = serial_obj
+        self.publisher = rospy.Publisher(topic, String, queue_size=10)
+
+    def run(self):
+        # TODO: Is it good to use rospy.Rate here? Or should we just add a timeout
```

Comment:
As discussed, I think rospy.Rate will be fine here. That way the node's loop is controlled. The serial library should be able to handle this, and ROS will be happy

Reply:
Discussed.